pkg/rpc/heartbeat.go

```diff
@@ -56,6 +60,31 @@ type HeartbeatService struct {
 	testingAllowNamedRPCToAnonymousServer bool
 }
 
+func checkClusterName(disablePeerNameVerification bool, clusterName string, peerName string) error {
+	if disablePeerNameVerification && (clusterName == "" || peerName == "") {
```

Comment:
Don't you want to OR the local and remote `disablePeerNameVerification`?
I'm thinking of the following situation:
- node A has flags with `--cluster-name=foo` and `--disable-peer-name-verification`
- node B has not had the new flags applied yet

node A is restarted as part of the name rollout.
node B restarts on its own, but still has not had the new flags applied.

At this point, node B will ping node A, but will fail due to `disablePeerNameVerification = false`. This will go into the `clusterName != peerName` check and fail.

Reply:
The OR is there, performed by the caller.
It's in the RFC and *also* checked in the unit test in the same package.

Reply:
oops. My bad. I had it in my head that `checkClusterName`'s signature was `(localDisable, remoteDisable, localName, remoteName)`. Sorry about that.

Reply:
np. so good to go?

Reply:
I'm good, but I'd like @bdarnell to take a look.